toffy/mph_comp.py

```diff
@@ -0,0 +1,180 @@
+    except FileNotFoundError:
+        raise FileNotFoundError(f"The FOV name supplied doesn't have a JSON file: {fov}")
+
+    count = count_dict[fov]
+    time = get_estimated_time(bin_file_dir, fov)
+
+    out_df = pd.DataFrame({
+        'fov': [fov],
+        'MPH': [median],
+        'total_count': [count],
+        'time': [time]})
+
+    # saves individual .csv files to csv_dir
+    out_df.to_csv(os.path.join(csv_dir, pulse_height_file), index=False)
+
+
+def combine_mph_metrics(csv_dir, return_data=False):
+    """Combines data from individual csvs into one
+        Args:
+            csv_dir (str): path where FOV mph data csvs are stored
+            return_data (bool): whether to return dataframe with mph metrics, default False
+
+        Returns:
+            combined mph data for all FOVs
+            """
+
+    # path validation checks
+    io_utils.validate_paths(csv_dir)
+
+    # for each csv retrieve mph values
+    fov_files = io_utils.list_files(csv_dir, "-mph_pulse.csv")
+    combined_rows = []
+    for i, file in enumerate(fov_files):
+        combined_rows.append(pd.read_csv(os.path.join(csv_dir, file)))
+
+    # calculate cumulative sums of total counts and time
+    combined_df = pd.concat(combined_rows)
+    run_order = []
+    for index, row in combined_df.iterrows():
+        run_order.append(int(''.join(filter(str.isdigit, row['fov']))))
+    combined_df['run_order'] = run_order
+    combined_df = combined_df.sort_values(by=['run_order'])
+    combined_df['cum_total_count'] = combined_df['total_count'].cumsum()
+    combined_df['cum_total_time'] = combined_df['time'].cumsum()
+    combined_df = combined_df.drop(columns=['run_order'])
+
+    # save csv to csv_dir
+    file_path = os.path.join(csv_dir, 'total_count_vs_mph_data.csv')
+    if os.path.exists(file_path):
+        os.remove(file_path)
+    combined_df.to_csv(file_path, index=False)
+
+    # return data
+    if return_data:
+        return combined_df
+
+
+def visualize_mph(mph_df, out_dir, regression: bool = False):
+    """Create a scatterplot visualizing median pulse heights by FOV cumulative count
+        Args:
+            mph_df (pd.DataFrame): data detailing total counts and pulse heights
+            out_dir (str): path of directory to save plot to
+            regression (bool): whether to plot regression line, default is False
+            """
+
+    # path validation checks
+    if out_dir is not None:
+        io_utils.validate_paths(out_dir)
+
+    # visualize the median pulse heights
+    plt.style.use('dark_background')
+    # plt.title('FOV total counts vs median pulse height')
+    fig = plt.figure()
+    ax1 = fig.add_subplot(111)
+    x = mph_df['cum_total_count']/1000000
+    y = mph_df['MPH']
+    ax1.set_xlabel('FOV cumulative count (in millions)')
+    ax1.set_ylabel('median pulse height')
+    ax1.scatter(x, y)
+    ax2 = ax1.twiny()
+    ax2.set_xlabel('estimated time (hours)')
+
+    # create time axis
```

Comment:
The tick-generation code can be a separate helper function. This will make it possible to test it, and declutter the main plotting function